init.go

```diff
@@ -0,0 +1,20 @@
+package ipldgit
+
+//go:generate go run ./gen .
+//go:generate go fmt ./
+
+import (
+	"github.com/ipfs/go-cid"
+	"github.com/ipld/go-ipld-prime"
+	mc "github.com/ipld/go-ipld-prime/multicodec"
+)
+
+var (
+	_ ipld.Decoder = Decode
+	_ ipld.Encoder = Encode
+)
+
+func init() {
+	mc.RegisterEncoder(cid.GitRaw, Encode)
+	mc.RegisterDecoder(cid.GitRaw, Decode)
+}
```

Comment:
Is this necessary for functioning, or just a convenience function for updating the global registry via module imports?

@warpfork I recall your aversion to global registries. Does encouraging codec creators to do this, but codec users to use custom LinkSystems when needed seem like a reasonable compromise?

Reply:
i believe this was the compromise we ended up at, yes

Reply:
Yep, that's exactly it.

Global registries have downsides, but using package-init-time's natural synchronization is just too valuable.  And having ways to opt-out of using the global registry, if some really needs to, seems to mitigate the biggest downsides handily.  So: :+1: